packet/gtp.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2017 Intel Corporation.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This file highly assumes that all packets in each particular situation
+// will have the same structure. Parsing, Encapsulating, Decapsulating
+// functions are base variants without any checkings which can be
+// painlessly changed for exact situation.
+
+package packet
+
+import (
+	"fmt"
+	"unsafe"
+
+	. "github.com/intel-go/yanff/common"
+)
+
+// GTPv1-U and GTPv1-C
+type GTPHdr struct {
+	HeaderType          uint8 // version, protocol type, extension header/sequence number/N-PDU flags
+	MessageType         uint8
+	MessageLength       uint16
+	TEID                uint32
+	SequenceNumber      uint16 // this is valid only with sequence number flag
+	NPDUNumber          uint8  // this is valid only with N-PDU flag
+	NextExtensionHeader uint8  // this is valid only with exatension header flag
+}
+
+// TODO current support of GTP-U: only two extension headers
+// NextExtensionHeader values
+const (
+	NoExtensionHeaders             = 0x00
+	UDPPortExtensionHeader         = 0x40
+	PDCP_PDU_NumberExtensionHeader = 0xc0
+)
+
+type UDPPort struct {
+	Lenght              uint8 // in 4 octets, here always 0x01
+	UDPPortNumber       uint16
+	NextExtensionHeader uint8
+}
+
+type PDCP_PDU_Number struct {
+	Length              uint8 // in 4 octets, here always 0x01
+	PDCP_PDU_Number_1   uint8
+	PDCP_PDU_Number_2   uint8
+	NextExtensionHeader uint8
+}
+
+// TODO for GTP-U only
+// MessageType values
+const (
+	// Signalling messages: path management
+	EchoRequest                           = 1
+	EchoResponse                          = 2
+	SupportedExtensionHeadersNotification = 31
+	// Signalling messages: tunnel management
+	ErrorIndication = 26
+	EndMarker       = 254
+	// G-PDU message = encapsulated user data
+	G_PDU = 255
+)
+
+// TODO Add "Information Element@ constants
```

Comment:
What does comment want to say?

Reply:
That we need to add "Information Element" constants which are defined in protocol RFCs